xtask/src/hlsl.rs

```diff
@@ -0,0 +1,85 @@
+use std::{
+    fs, io,
+    path::{Path, PathBuf},
+};
+
+use miette::{ensure, Diagnostic, SourceOffset, SourceSpan};
+use serde::Deserialize;
+
+#[derive(Debug, Diagnostic, thiserror::Error)]
+#[error("failed to parse YAML configuration at path {}", file_name.display())]
+struct BadYamlParse {
+    #[diagnostic(source_code)]
+    file_name: PathBuf,
+    #[diagnostic(span)]
+    location: Option<SourceSpan>,
+    #[diagnostic(source)]
+    source: BadYamlParseKind,
+}
+
+#[derive(Debug, thiserror::Error)]
+enum BadYamlParseKind {
+    #[error(transparent)]
+    Read { source: io::Error },
+    #[error(transparent)]
+    Parse { source: serde_yaml::Error },
+    #[error("no configuration was specified")]
+    Empty,
+}
+
+#[derive(Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct Config {
+    pub vertex: Option<Vec<ConfigItem>>,
+    pub fragment: Option<Vec<ConfigItem>>,
+    pub compute: Option<Vec<ConfigItem>>,
+}
```

Comment:
todo: Auto-generate YAML from `tests/snapshots.rs`, instead of Bash variables lists.